code/modules/wiremod/components/ntnet/ntnet_receive.dm

```diff
@@ -19,31 +19,29 @@
 	/// Encryption key
 	var/datum/port/input/enc_key
 
-/obj/item/circuit_component/ntnet_receive/Initialize(mapload)
-	. = ..()
-	init_network_id(__NETWORK_CIRCUITS)
-
 /obj/item/circuit_component/ntnet_receive/populate_options()
 	list_options = add_option_port("List Type", GLOB.wiremod_basic_types)
 
 /obj/item/circuit_component/ntnet_receive/populate_ports()
 	data_package = add_output_port("Data Package", PORT_TYPE_LIST(PORT_TYPE_ANY))
 	enc_key = add_input_port("Encryption Key", PORT_TYPE_STRING)
-	RegisterSignal(src, COMSIG_COMPONENT_NTNET_RECEIVE, PROC_REF(ntnet_receive))
+	RegisterSignal(SSdcs, COMSIG_GLOB_CIRCUIT_NTNET_DATA_SENT, PROC_REF(ntnet_receive))
 
 /obj/item/circuit_component/ntnet_receive/pre_input_received(datum/port/input/port)
 	if(port == list_options)
 		var/new_datatype = list_options.value
 		data_package.set_datatype(PORT_TYPE_LIST(new_datatype))
 
 
-/obj/item/circuit_component/ntnet_receive/proc/ntnet_receive(datum/source, datum/netdata/data)
+/obj/item/circuit_component/ntnet_receive/proc/ntnet_receive(obj/item/circuit_component/ntnet_send/source, list/data)
 	SIGNAL_HANDLER
 
-	if(data.data["enc_key"] != enc_key.value)
+	if(!SSmodular_computers.check_function())
```

Comment:
You need to move this off of `modular_computers`, I still don't think it makes any sense

Reply:
I forgot to, I'll get to it tomorrow (I also can't find the draft button on git mobile)